admin/aws.py

```diff
@@ -397,6 +401,11 @@ class FakeAWS(object):
     """
     def __init__(self):
         self.cloudfront_invalidations = []
+        for bucket in self.s3_buckets.values():
+            for key, contents in bucket.items():
+                if isinstance(contents, basestring):
```

Comment:
`basestring` always makes me wary.  Does the real API really work correctly with either bytes or unicode?  Or does it work with some unicode some of the time by accident?  I suspect this should really only allow bytes here but if both bytes and unicode are meant to work, I'd suggest going with `(bytes, unicode)` to be explicit about it.

Alternatively, the values of this dictionary should be a PRecord (or characteristic) thing instead of a tuple, and then this could just be `not isinstance(contents, TypedContents)` instead.

Explicitly rejecting unicode here if unicode isn't how the real AWS APIs work might be valuable, though.